src/vs/workbench/contrib/preferences/browser/settingsTreeModels.ts

```diff
@@ -291,10 +291,12 @@ export class SettingsTreeSettingElement extends SettingsTreeElement {
 	}
 
 	matchesAnyFeature(featureFilters?: Set<string>): boolean {
-		const features = ['explorer', 'search', 'debug', 'scm', 'extensions', 'terminal', 'task', 'problems', 'output', 'comments', 'remote', 'timeline'];
+		const features = tocData.children!.filter(child => child.id === 'features')[0].children!.map(child => child.id.substring(9));
+
 		if (!featureFilters || !featureFilters.size) {
 			return true;
 		}
+
 		return Array.from(featureFilters).filter(feature => features.includes(feature)).some(feature => this.setting.key.toLowerCase().startsWith(feature));
```

Comment:
What I have a problem with is `this.setting.key.toLowerCase().startsWith`, this actually isn't sufficient if you see what patterns are included in settingsLayout.ts

Reply:
oh. I see.

Reply:
resolved, but now when you search @feature:debug, is returning some extension settings as well which is not ideal

Reply:
that is now resolved as well by checking if !extensionInfo on the setting within matchesAnyFeatures